### PR TITLE
Update DR module for KFD v1.24 release

### PR DESCRIPTION
**Pull request by lnovara**
TODOs:
- [x] Update Velero and its plugins
- [x] Check required Terraform and providers versions in Terraform modules
- [x] Add CI support for Kubernetes v1.24
- [x] Add v1.10 release notes

### CODE REVIEW

**Comment by lnovara** [comment]
Closes #49

File: README.md [hunk 1]
```diff
@@ -63,16 +63,15 @@ Deploy the necessary infrastructure to persist the backups natively in cloud pro
...
 
 ## Compatibility
 
 | Kubernetes Version |   Compatibility    |                        Notes                        |
 | ------------------ | :----------------: | --------------------------------------------------- |
-| `1.20.x`           | :white_check_mark: | No known issues                                     |
 | `1.21.x`           | :white_check_mark: | No known issues                                     |
 | `1.22.x`           | :white_check_mark: | No known issues                                     |
-| `1.23.x`           |     :warning:      | Conformance tests passed. Not officially supported. |
+| `1.23.x`           |     :warning:      | No known issues                                     |
```
Comment (nutellinoit) [inline README.md:73]: Why ⚠️  on 1.23 and 1.24?

File: docs/COMPATIBILITY_MATRIX.md [hunk 2]
```diff
@@ -1,20 +1,21 @@
...
+| v1.9.1                              |                    |                    |                    |                    |                    |                    | :x:                | :x:                | :x:                | :x:                |                   |
+| v1.9.2                              |                    |                    |                    |                    |                    |                    | :warning:          | :warning:          | :warning:          | :warning:          |                   |
+| v1.9.3                              |                    |                    |                    |                    |                    |                    | :white_check_mark: | :white_check_mark: | :white_check_mark: | :warning:          |                   |
+| v1.10.0                             |                    |                    |                    |                    |                    |                    |                    | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark |
```
Comment (nutellinoit) [inline docs/COMPATIBILITY_MATRIX.md:18]:
```suggestion
| v1.10.0                             |                    |                    |                    |                    |                    |                    |                    | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: |
```

File: docs/releases/v1.10.0.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+| `velero-plugin-for-csi`             | [`v0.3.1`](https://github.com/vmware-tanzu/velero-plugin-for-csi/releases/tag/v0.3.1)             | `v0.2.0`         |
+
+> Please refer the individual release notes to get a detailed info on the
+> releases.
+
+## Features 💥
+
+- Updated Velero to v1.9.2
+- Removed the deprecated `eks-velero` Terraform module
+- Added support for Kubernetes v1.14.x
```
Comment (nutellinoit) [inline docs/releases/v1.10.0.md:27]:
```suggestion
- Added support for Kubernetes v1.24.x
```

**Review by nutellinoit — CHANGES_REQUESTED** [review]
LGTM with some minor fixes!

**Review by nutellinoit — APPROVED** [review]
LGTM 🎉

File: katalog/velero/velero-on-prem/README.md [hunk 4]
```diff
@@ -1,11 +1,11 @@
 # Velero on Premises
 
 This Velero deployment is ready to be deployed in any Kubernetes cluster as it includes a MinIO instance compatible with
-the [AWS Velero plugin](https://github.com/vmware-tanzu/velero-plugin-for-aws/tree/v1.1.0).
+the [AWS Velero plugin](https://github.com/vmware-tanzu/velero-plugin-for-aws/tree/v1.5.1).
 
 ## Image repository and tag
 
-- Velero AWS Plugin image: `velero/velero-plugin-for-aws:v1.2.0`
+- Velero AWS Plugin image: `velero/velero-plugin-for-aws:v1.5.1`
```
Comment (ralgozino) [inline katalog/velero/velero-on-prem/README.md:8]: I don't think we use the AWS Plugin for on-prem, do we?

**Review by ralgozino — CHANGES_REQUESTED** [review]
Overall the PR is looking good, some minor styling changes and the following two things:

1. It seems that the on-prem part has some stuff that has been copied over from AWS but is not used.
2. Can we add at least a maintenance.md at the `velero` level with some basic instructions? We can improve it later.

Thank you!

File: katalog/velero/velero-on-prem/README.md (hunk 4 above)
Comment (lnovara) [reply katalog/velero/velero-on-prem/README.md:8]: We do because for on-prem we rely on MinIO to expose an S3 compliant interface for Velero to use.

**Comment by lnovara** [comment]
Added maintenance instructions, thanks @ralgozino!

**Review by ralgozino — APPROVED** [review]
LG(reat)TM 🚀

**Pull request merged**